src/main/java/edu/ie3/datamodel/io/factory/timeseries/CosmoTimeBasedWeatherValueFactory.java

```diff
@@ -72,14 +78,22 @@ protected TimeBasedValue<WeatherValue> buildModel(TimeBasedWeatherValueData data
         data.getQuantity(WIND_DIRECTION, StandardUnits.WIND_DIRECTION);
     ComparableQuantity<Speed> windVelocity =
         data.getQuantity(WIND_VELOCITY, StandardUnits.WIND_VELOCITY);
+    ComparableQuantity<Temperature> groundTemp0cm =
+        data.getQuantityOptional(GROUND_TEMPERATURE_0CM, StandardUnits.TEMPERATURE).orElse(null);
+    ComparableQuantity<Temperature> groundTemp80cm =
+        data.getQuantityOptional(GROUND_TEMPERATURE_80CM, StandardUnits.TEMPERATURE).orElse(null);
+
```

Comment:
```suggestion
   Optional< ComparableQuantity<Temperature>> groundTemp0cm =
        data.getQuantityOptional(GROUND_TEMPERATURE_0CM, StandardUnits.TEMPERATURE);
    Optional< ComparableQuantity<Temperature>> groundTemp80cm =
        data.getQuantityOptional(GROUND_TEMPERATURE_80CM, StandardUnits.TEMPERATURE);
```

Let's go with optionals here